crates/ruff_python_semantic/src/branches.rs

```diff
@@ -0,0 +1,52 @@
+use std::ops::Index;
+
+use ruff_index::{newtype_index, IndexVec};
+
+/// ID uniquely identifying a branch in a program.
+///
+/// For example, given:
+/// ```python
+/// if x > 0:
+///     pass
+/// elif x > 1:
+///     pass
+/// else:
+///     pass
+/// ```
+///
+/// Each of the three arms of the `if`-`elif`-`else` would be considered a branch, and would be
+/// assigned their own unique [`BranchId`].
+#[newtype_index]
```

Comment:
i don't like that the macro adds the fields to the struct, it makes it hard to follow and hard to IDE/tool-inspect what data that struct actually is